equinix/common_test.go

```diff
@@ -0,0 +1,120 @@
+package equinix
+
+import (
+	"fmt"
+	"strings"
+	"time"
+)
+
+// list of plans and metros and os used as filter criteria to find available hardware to run tests
+var (
+	preferable_plans  = []string{"x1.small.x86", "t1.small.x86", "c2.medium.x86", "c3.small.x86", "c3.medium.x86", "m3.small.x86"}
+	preferable_metros = []string{"ch", "ny", "sv", "ty", "am"}
+	preferable_os     = []string{"ubuntu_20_04"}
+)
+
+// Deprecated: use the identical TestDeviceTerminationTime from internal/acceptance instead
+func testDeviceTerminationTime() string {
+	return time.Now().UTC().Add(60 * time.Minute).Format(time.RFC3339)
+}
+
+// This function should be used to find available plans in all test where a metal_device resource is needed.
+//
+// TODO consider adding a datasource for equinix_metal_operating_system and making the local.os conditional
+//
+//	https://github.com/equinix/terraform-provider-equinix/pull/220#discussion_r915418418equinix_metal_operating_system
+//	https://github.com/equinix/terraform-provider-equinix/discussions/221
+func confAccMetalDevice_base(plans, metros, os []string) string {
+	return fmt.Sprintf(`
```

Comment:
Perhaps this was a flake that will go away following your latest push, but I noticed this from the previous test run:
```
=== NAME  TestAccDataSourceMetalDevices
    list_datasource_test.go:15: Step 1/1 error: Error running pre-apply plan: exit status 1
        
        Error: Unable to load records: unexpected EOF
        
          with data.equinix_metal_plans.test,
          on terraform_plugin_test.tf line 17, in data "equinix_metal_plans" "test":
          17: data "equinix_metal_plans" "test" {
        
--- FAIL: TestAccDataSourceMetalDevices (63.95s)
```

Reply:
That error is from the plans datasource, so shouldn't be impacted by changes to the device resource/datasource code, but we'll see soon enough if it's intermittent or not.